pkg/policycache/informer.go

```diff
@@ -95,14 +108,59 @@ func (c *Controller) deleteNsPolicy(obj interface{}) {
 
 // Run waits until policy informer to be synced
 func (c *Controller) Run(workers int, stopCh <-chan struct{}) {
+
 	logger := c.log
 	logger.Info("starting")
 	defer logger.Info("shutting down")
 
-	if !cache.WaitForCacheSync(stopCh, c.pSynched) {
-		logger.Info("failed to sync informer cache")
-		return
+	if !cache.WaitForCacheSync(stopCh, c.pSynched, c.nspSynched) {
+		logger.Error(nil, "Failed to sync informer cache")
+		os.Exit(1)
 	}
 
 	<-stopCh
 }
+
+// CheckPolicySync wait until the internal policy cache is fully loaded
+func (c *Controller) CheckPolicySync(stopCh <-chan struct{}) {
+
+	if !cache.WaitForCacheSync(stopCh, c.pSynched, c.nspSynched) {
+		logger.Error(nil, "Failed to sync informer cache")
+		os.Exit(1)
```

Comment:
Since we have this check here, do we need it in `func (c *Controller) Run(workers int, stopCh <-chan struct{})`?

Reply:
done